lib/p2p/NodeList.ts

```diff
@@ -39,10 +40,26 @@ class NodeList extends EventEmitter {
   /**
    * Check if a node with a given nodePubKey exists.
    */
-  public has = (nodePubKey: string) => {
+  public has = (nodePubKey: string): boolean => {
     return this.nodes.has(nodePubKey);
   }
 
+  public remove = (nodePubKey: string): void => {
+    this.nodes.delete(nodePubKey);
+  }
+
+  public inBannedList = (nodePubKey: string): boolean  => {
+    return this.bannedNodes.has(nodePubKey);
+  }
+
+  public addBannedList = (nodePubKey: string): void  => {
+    this.bannedNodes.add(nodePubKey);
+  }
+
+  public removeBannedList = (nodePubKey: string): void  => {
```

Comment:
Similarly for these remove and add banned list methods, I think it makes most sense just to add/remove to the list when we're banning/unbanning nodes. In other words, we wouldn't want to remove or add to the list unless we're actually banning or unbanning.